edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/upstream/BrokeredCloudConnectionProvider.cs

```diff
@@ -27,10 +27,33 @@ public Task<Try<ICloudConnection>> Connect(IClientCredentials clientCredentials,
             return this.Connect(clientCredentials.Identity, connectionStatusChangedHandler);
         }
 
-        public Task<Try<ICloudConnection>> Connect(IIdentity identity, Action<string, CloudConnectionStatus> connectionStatusChangedHandler)
+        public async Task<Try<ICloudConnection>> Connect(IIdentity identity, Action<string, CloudConnectionStatus> connectionStatusChangedHandler)
```

Comment:
Are these changes related to the config missing?

Reply:
The problem it solves came out testing the missing twin problem. The original problem was that when a new child edge device is starting up (so it has no stored twins) and the parent is not ready (so it cannot acquire the twin), then this problems happened (this PR fixes that problem). A secondary problem was that sometimes during startup it happens that edgeHub starts executing upstream operation when the infrastructure is not ready yet. It polluted the logs with error messages quite often, so adding this extra check for the connection status mitigates this.